pkg/controller/pagerdutyintegration/clusterdeployment_deleted.go

```diff
@@ -143,6 +164,15 @@ func (r *ReconcilePagerDutyIntegration) handleDelete(pdclient pd.Client, pdi *pa
 		r.reqLogger.Error(err, "Error deleting Secret", "Namespace", cd.Namespace, "Name", secretName)
 	}
 
+	// switch the PD syncset to Upsert prior to deletion if the clusterdeployment is relocating
+	if utils.HasAnnotation(cd, hiveconstants.RelocateAnnotation, strings.HasSuffix, string(hivev1.RelocateOutgoing), string(hivev1.RelocateComplete)) {
+		r.reqLogger.Info("Updating PD SyncSet", "Namespace", cd.Namespace, "Name", secretName)
+		err = utils.UpdateSyncSetApplyMode(secretName, cd.Namespace, r.client, hivev1.UpsertResourceApplyMode, r.reqLogger)
```

Comment:
Why need to change the apply mode here?

Reply:
The PD SyncSets use the `Sync` apply mode - the problem with that is that it means that when the SyncSet is deleted by the outgoing operator, Hive will remove the resource it manages on the remote cluster - in this case the PD secret in `openshift-monitoring`. That leaves the cluster in a state where it is potentially not able to report to PD.

So my strategy for relocation has been to change the apply mode to `Upsert` prior to the SyncSet's deletion on the outgoing Hive shard, so that when the syncset is deleted it results in nil impact to the destination cluster.